### ISSUE TITLE
Allow passing in custom TLS config to manager for use with webhooks

### ISSUE THREAD

**Issue by alastor-erinyes** [kubernetes-sigs/controller-runtime#852]
Currently we allow specifying the certificate directory with the manager options, but this can be too restrictive in certain usecases.

I propose we allow specifying the entire `tls.Config{}` object while still preserving the current certDir option.  This gives security configuration for webhooks much more flexibility than the current model.

In `pkg/webhook/server.go:` the `tls.Config` file is instantiated as the following:
```
        cfg := &tls.Config{
		NextProtos:     []string{"h2"},
		GetCertificate: certWatcher.GetCertificate,
	}
```
Instead just allow us to specify this entire config via `manager.Options{}`

Something like:
```
manager.Options{
    WebhookTLSConfig: &tls.Config{},
}
```
If this is not specified, then we fall back to reading certs from the configured directory.

**Comment by maxsmythe** [comment]
If this is implemented, it be nice if the `GetCertificate` function was public so that it could be included in the custom TLS config.

That way users could still get the benefit of the io-watching code.

**Comment by sysword** [comment]
I wonder why dont we use restConfig.TLSClientConfig to do authentication, is that enough?

**Comment by grim-fendango** [comment]
This would be useful for webhooks that require zero down time during certificate rotation as it would allow us to have an A and B certificate.

**Comment by huornlmj** [comment]
Would this provide the capability for users to disable TLS 1.0 and TLS 1.1?

**Comment by ahalim-intel** [comment]
I think this is an important feature that manager allows to pass custom TLS config if needed for example configuring webhook with mTLS and restricting to run on more secured version of TLS.

#1152 would solve this issue if re-opened.

**Comment by RainbowMango** [comment]
cc @vincepri @alvaroaleman @DirectXMan12

**Comment by umangachapagain** [comment]
There were multiple attempts to fix this issue: #853 #1152 #1623 . All closed by stale bot.
But, there was no objection to the idea.

Do we still want/need this? @vincepri @joelanford @alvaroaleman 
If yes, I can try to get it over the finish line. If not, we should close this.

**Comment by huornlmj** [comment]
Yes!

**Comment by akalenyu** [comment]
> Was this solved via #1897 or is something missing? cc @akalenyu

I believe this is solved now as we allow full control over the TLS config used by the webhook server

**Comment by alvaroaleman** [comment]
thanks again!
